src/RSocketRequester.cpp

```diff
@@ -26,14 +26,6 @@ RSocketRequester::~RSocketRequester() {
   }
 }
 
-void RSocketRequester::closeSocket() {
```

Comment:
I want us to clear this confusion up and solve this for real. I suggest something like this:
```cpp
struct RSocket {
  // emphasis on non-owning
  virtual void RSocketRequester& getRemote();
  virtual void disconnect() = 0;
  virtual void close() = 0;
};

struct RSocketClient : public RSocket {
  virtual Future resume();
};

// if it doesn't need extra methods, can also be a hard typedef
using RSocketServer = RSocket; 
```